services/web/server/src/simcore_service_webserver/scicrunch/service_client.py

```diff
@@ -0,0 +1,189 @@
+        self.settings = settings
+        self.client = client
+        self.base_url = URL.build(
+            scheme=self.settings.api_base_url.scheme,
+            host=self.settings.api_base_url.host,
+        )
+
+    # Website links ---------
+
+    def get_portal_link(self) -> str:
+        return str(self.base_url.with_path("/resources/"))
+
+    def get_rrid_link(self, rrid: str) -> str:
+        # NOTE: for some reason scicrunch query does not like prefix!
+        return str(
+            self.base_url.with_path("/resources/Any/search").with_query(
+                q="undefined", l=rrid.replace("RRID: ", "").strip()
+            )
+        )
+
+    def get_new_submission_link(self) -> str:
+        return str(self.base_url.with_path("/resources/about/resource"))
+
+    # Application instance ---------
+
+    @classmethod
+    def acquire_instance(
+        cls, app: MutableMapping[str, Any], settings: SciCrunchSettings
+    ) -> "SciCrunchAPI":
+        """ Returns single instance for the application and stores it """
+        obj = cls.get_instance(app)
+        if obj is None:
+            session = get_client_session(app)
+            app[f"{__name__}.SciCrunchAPI"] = obj = cls(session, settings)
+        return obj
+
+    @staticmethod
+    def get_instance(
+        app: MutableMapping[str, Any], *, raises=False
+    ) -> Optional["SciCrunchAPI"]:
+        """ Get's application instance """
+        obj = app.get(f"{__name__}.SciCrunchAPI")
+        if raises and obj is None:
+            raise web.HTTPServiceUnavailable(
+                reason="Link to scicrunch.org services is currently disabled."
+            )
+        return obj
+
+    # API calls --------
+
+    @classmethod
+    def validate_identifier(cls, rrid: str) -> str:
+        match = cls.RRID_RE.match(rrid.strip())
+        if match:
+            rrid = match.group(2)  # WARNING: captures indexing is 1-based
+            assert rrid and isinstance(rrid, str), "Captured {rrid}"  # nosec
+            return rrid
+
+        raise web.HTTPUnprocessableEntity(reason=f"Invalid format for an RRID '{rrid}'")
+
+    async def validate_resource(self, rrid: str) -> ValidationResult:
+        try:
+            rrid = self.validate_identifier(rrid)
+            versions = await get_all_versions(rrid, self.client, self.settings)
+            return ValidationResult.VALID if versions else ValidationResult.INVALID
+
+        except aiohttp.ClientResponseError as err:
+            if err.status == HTTPStatus.BAD_REQUEST:
+                return ValidationResult.INVALID
+            return ValidationResult.UNKNOWN
+
+        except aiohttp.ClientError:
+            # connection handling and server response misbehaviors
+            # ClientResposeError over 300
+            # raises? --> does NOT mean it is invalid but that cannot determine validity right now!!!
+            # - server not reachable: down, wrong address
+            # - timeout: server slowed down (retry?)
```

Comment:
I would log a warning here for easier debugging.

Reply:
these were thoughts during design ... i will clean up